### PR TITLE
fix compilation of trace CTF

### PR DESCRIPTION
**Pull request by ernstblechaPT**
#198 broke the interface used by the trace CTF implementation

### CODE REVIEW

File: src/core/basicfb.cpp [hunk 1]
```diff
@@ -195,12 +204,16 @@ void CBasicFB::traceInstanceData() {
...
 
-  for(TPortId i = 0; i < internalFbs.size(); ++i) {
-    CFunctionBlock *value = getInternalFB(i);
-    std::string &valueString = internalFbs[i];
-    valueString.reserve(value->getToStringBufferSize());
-    value->toString(valueString.data(), valueString.capacity());
-    internalFbs_c_str[i] = valueString.c_str();
+  TPortId i = 0;
+  for(TFBContainerList::iterator it(getChildren().begin()); it != getChildren().end(); ++it){
+    if((*it)->isFB()) {
```
Comment (mx990) [inline src/core/basicfb.cpp]: A basic FB can only contain internal FBs (or adapters), both of which happen to return `true` for `isFB()`, so I would suggest to simply use `getChildren()` as-is.

**Review by mx990 — CHANGES_REQUESTED** [review]

File: src/core/basicfb.cpp [hunk 2]
```diff
@@ -160,12 +160,19 @@ size_t CBasicFB::getToStringBufferSize() const {
...
+  }
+  return i;
+}
 
+void CBasicFB::traceInstanceData() {
   std::vector<std::string> inputs(mInterfaceSpec->mNumDIs);
   std::vector<std::string> outputs(mInterfaceSpec->mNumDOs);
   std::vector<std::string> internals(cmVarInternals ? cmVarInternals->mNumIntVars : 0);
-  std::vector<std::string> internalFbs(getInternalFBNum());
+  std::vector<std::string> internalFbs(internalFBCount(this));
```
Comment (mx990) [inline src/core/basicfb.cpp]: I would suggest simply using `getChildren().size()` here.

File: src/core/basicfb.cpp [hunk 3]
```diff
@@ -195,12 +202,14 @@ void CBasicFB::traceInstanceData() {
     internals_c_str[i] = valueString.c_str();
   }
 
-  for(TPortId i = 0; i < internalFbs.size(); ++i) {
-    CFunctionBlock *value = getInternalFB(i);
+  TPortId i = 0;
+  for(TFBContainerList::iterator it(getChildren().begin()); it != getChildren().end(); ++it){
```
Comment (mx990) [inline src/core/basicfb.cpp]: I would suggest either iterating from `0` to `getChildren().size()` or using a [range-for](https://en.cppreference.com/w/cpp/language/range-for) loop.

**Review by mx990 — CHANGES_REQUESTED** [review]

**Review by mx990 — APPROVED** [review]

**Comment by ernstblechaPT** [comment]
seems like there still is a problem - need to look into that

appears to be this problem: https://github.com/eclipse-4diac/4diac-forte/pull/198#issuecomment-2266698110

**Pull request merged**